share/test/schemas/RecordWithRequiredFields.avsc

```diff
@@ -0,0 +1,53 @@
+{
+  "type" : "record",
+  "name" : "RecordWithRequiredFields",
+  "namespace" : "org.apache.avro",
+  "fields" : [ {
+    "name" : "f0",
```

Comment:
IMO it would be easier to understand the test cases if `fN` fields were named after their types, e.g. `stringField` and `enumField`.
Since this PR makes it easier to pinpoint the problematic field I see no problem in using `stringField` several types in the schema.

Reply:
will do